### PR TITLE
Add event when add/del team member

### PR DESCRIPTION
**Pull request by YglesEyes**

### CODE REVIEW

File: core/server/guilds.js [hunk 1]
```diff
@@ -79,27 +102,35 @@ Meteor.methods({
...
     if (!this.userId) throw new Meteor.Error('not-authorized', 'User not allowed');
 
-    if (!canEditGuild(Meteor.user(), Guilds.findOne(guildId))) throw new Meteor.Error('not-authorized', `Missing permissions to edit team members`);
+    const guild = Guilds.findOne(guildId);
+    if (!canEditGuild(Meteor.user(), guild)) throw new Meteor.Error('not-authorized', `Missing permissions to edit team members`);
 
-    const user = Meteor.users.findOne(userId);
-    if (user.guildId !== guildId) throw new Meteor.Error('user-invalid', 'Given user is not in the team');
+    let users = Meteor.users.find({ _id: { $in: userIds }, guildId });
+    if (users.length === 0) throw new Meteor.Error('user-invalid', 'Given user is not in the team');
```
Comment (Donorhan) [inline core/server/guilds.js:120]: This part is wrong, in fact in the previous code it was checked that the deleted user was in the team.But I wonder if this check is really necessary 🤔

File: core/server/guilds.js [hunk 2]
```diff
@@ -79,27 +102,35 @@ Meteor.methods({
...
-    Meteor.users.update(userId, { $unset: { guildId: 1 } });
+    Meteor.users.update({ _id: { $in: userIds } }, { $unset: { guildId: 1 } });
 
     // analytics
-    analytics.identify(Meteor.users.findOne(userId));
-    analytics.track(this.userId, '👨‍👨‍👦 Guild Remove User', { user_id: userId, guild_id: guildId });
-    analytics.updateGuild(Guilds.findOne(guildId), {}, userId);
+    users = Meteor.users.find({ _id: { $in: userIds } }).fetch();
+    users.forEach(currentUser => {
+      analytics.identify(Meteor.users.findOne(currentUser));
```
Comment (Donorhan) [inline core/server/guilds.js:127]: You can use the `currentUser` variable here and rename it to `removedUser` maybe

File: core/server/guilds.js [hunk 3]
```diff
@@ -47,6 +47,29 @@ const createGuild = params => {
...
 
+
+/**
+ * Create a new guild and emit event new_guild
+ * @fires member_changed
+ * @param {string} _guildId - The modified guild ID
+ *
+ * @returns {void}
+ */
+const emitMemberEvent = _guildId => {
```
Comment (Donorhan) [inline core/server/guilds.js:58]: Wouldn't it be better to rename it to emitTeamMemberEvent?

File: core/server/guilds.js [hunk 4]
```diff
@@ -47,6 +47,29 @@ const createGuild = params => {
...
+  const guildMembers = Meteor.users.find({ guildId: _guildId }).fetch();
```
Comment (Donorhan) [inline core/server/guilds.js:59]: If you want to send the total only it is better to do :
`const guildMemberCount = Meteor.users.find({ guildId: _guildId }).count();`.

For later we could send the type of difference 'removed' or 'added' and the ids of the modified users for the moment it won't be very useful so no need.

File: core/client/ui/user-list-selection.js [hunk 5]
```diff
@@ -28,8 +28,12 @@ Template.userListSelection.events({
     toggleUserSelection(userId, templateInstance);
   },
   'click .js-submit'(event, templateInstance) {
-    Session.set('usersSelected', Object.keys(templateInstance.selectedUsers.get()));
-    closeModal();
+    const userList = Object.keys(templateInstance.selectedUsers.get());
+    const msg = Meteor?.settings?.public?.confirmMessage?.addUsers ? Meteor.settings.public.confirmMessage.addUsers : `Are you sure to add these users to the team?`;
```
Comment (Donorhan) [inline core/client/ui/user-list-selection.js:32]: A shorter version: 
```const msg = Meteor .settings .public .confirmMessage ?.addUsers || `Are you sure you want to add these users to the team?` ```

File: core/client/ui/team-settings-members.js [hunk 6]
```diff
@@ -32,15 +32,38 @@ Template.teamSettingsMembers.events({
...
+    event.preventDefault();
+    event.stopPropagation();
+
+    const { userId } = event.currentTarget.dataset;
+    toggleUserSelection(userId, templateInstance);
+  },
   'click .js-remove-team-user'() {
-    lp.notif.confirm('Team member deletion', `Are you sure to remove this user from the team?`, () => {
-      Meteor.call('removeTeamUser', Meteor.user().guildId, this.user._id, error => {
+    const msg = Meteor?.settings?.public?.confirmMessage?.delUsers ? Meteor.settings.public.confirmMessage.delUsers : `Are you sure to remove this user from the team?`;
```
Comment (Donorhan) [inline core/client/ui/team-settings-members.js:64]: A shorter version:
```
const msg = Meteor .settings .public .confirmMessage ?.delUsers || `Are you sure you want to remove these users to the team?`
```

**Review by Donorhan — CHANGES_REQUESTED** [review]

File: core/server/guilds.js (hunk 1 above)
Comment (YglesEyes) [reply core/server/guilds.js:120]: I think it's important if selected user is in another team we have to throw an error, but I have to change ```(users.length === 0)``` to ```(users.length === userIds.length)```

File: core/server/guilds.js (hunk 3 above)
Comment (YglesEyes) [reply core/server/guilds.js:58]: it's an internal and not exported function it's a little redondant no ? But it's ok

**Comment by YglesEyes** [comment]
I did some change, I let you see my answer and tell me if you are ok with us or not and I will do another commit if needed

**Pull request merged**